core/src/main/java/eu/mhutti1/utils/storage/adapter/StorageViewHolder.kt

```diff
@@ -39,7 +39,10 @@ internal class StorageViewHolder(
       else R.string.external_storage
     )
     file_size.text = storageCalculator.calculateAvailableSpace(item.file) + " / " +
-      storageCalculator.calculateTotalSpace(item.file)
-    containerView.setOnClickListener { onClickAction.invoke(item) }
+      storageCalculator.calculateTotalSpace(item.file) + "  "
+    file_name.setOnClickListener {
+      onClickAction.invoke(item)
+      file_name.toggle()
```

Comment:
Can you tell me why this is wrong and not conserving the checked state of the button, do I need to save the checked state of the button via sharedPrefs?

Reply:
the checked state of the buttons is already saved as the users preferred storage, I believe I left a comment to this effect. I'll investigate further as this is the last ticket before I can start release of 3.4.1

Reply:
Okay toggle doesn't work because the next time you see the toggle button it is not the same togglebutton. The dialog gets recreated and it is a totally new hierarchy with no knowledge of what happened previously.

We must control the visual state of the togglebutton by setting `isChecked`

```
file_name.isChecked = sharedPreferenceUtil.prefStorage == item.name
```